samples/Basic_Alexa/app/application.cpp

```diff
@@ -0,0 +1,170 @@
+	server.paths.setDefault(onHttpRequest);
+	server.setBodyParser(MIME_JSON, bodyToStringParser);
+	server.setBodyParser(MIME_XML, bodyToStringParser);
+
+	if(!UPnP::deviceHost.begin()) {
+		debug_e("UPnP initialisation failed");
+		return;
+	}
+
+	UPnP::deviceHost.registerDevice(&bridge);
+
+	/*
+	 * To avoid confusion when testing with a Host or real device, ensure lighting devices are
+	 * created with different names so they can be distinguished in the Alexa App.
+	 */
+	auto Name = [](unsigned id) -> String {
+		String s;
+		s += MACROQUOTE(SMING_ARCH);
+		s += '-';
+		s += id;
+		return s;
+	};
+
+	devices.addElement(new Hue::OnOffDevice(101, Name(101)));
+	devices.addElement(new Hue::DimmableDevice(102, Name(102)));
+	devices.addElement(new Hue::ColourDevice(103, Name(103)));
+
+	devices.addElement(new Hue::OnOffDevice(253, Name(253)));
+	devices.addElement(new Hue::DimmableDevice(254, Name(254)));
+	devices.addElement(new Hue::ColourDevice(255, Name(255)));
+
+	devices.addElement(new Hue::OnOffDevice(256, Name(256)));
+
+	devices.addElement(new Hue::OnOffDevice(500, Name(500)));
+	devices.addElement(new Hue::DimmableDevice(501, Name(501)));
+	devices.addElement(new Hue::ColourDevice(502, Name(502)));
+
+	devices.addElement(new Hue::ColourDevice(54321, Name(54321)));
+	devices.addElement(new Hue::ColourDevice(0x7FFFFFFF, Name(0x7FFFFFFF)));
+
+	// Connect the LED pin to light 101
+	digitalWrite(LED_PIN, HIGH); // Turn off - state is inverted
+	pinMode(LED_PIN, OUTPUT);
+
+	// Monitor when device attributes change
+	bridge.onStateChanged([&](const Hue::Device& device, Hue::Device::Attributes attr) {
+		debug_i("#%u %s changed [%s]", device.getId(), device.getName().c_str(), toString(attr));
+
+		/*
+		 * Control the LED pin
+		 */
+		using Attr = Hue::Device::Attribute;
+		if(device.getId() == 101 && attr[Attr::on]) {
+			unsigned value;
+			if(device.getAttribute(Attr::on, value)) {
+				digitalWrite(LED_PIN, value == 0);
+			}
+		}
+	});
+
+	// Monitor and handle requests to authorize or revoke users
+	bridge.onConfigChange([&](const Hue::Bridge::Config& config) {
+		debug_i("%s: deviceType=%s, name=%s",
+				config.type == Hue::Bridge::Config::Type::AuthorizeUser ? "Authorize" : "Revoke", config.deviceType,
+				config.name);
+
+		bridge.configure(config);
+	});
+
+	/*
+	 * Allow creation of users.
+	 *
+	 * WARNING! Security risk. In a real application, you would only enable pairing
+	 * on request from a secure web-page or by pressing a button. In addition, a suitable
+	 * timeout should be applied (e.g. 1 minute).
+	 *
+	 */
+	bridge.enablePairing(true);
+}
+
+} // namespace
+
+void init()
+{
+	Serial.setTxBufferSize(4096);
+	Serial.begin(SERIAL_BAUD_RATE);
+	Serial.systemDebugOutput(true);
+
+	WifiStation.enable(true, false);
+	WifiStation.config(WIFI_SSID, WIFI_PWD);
+	WifiAccessPoint.enable(false, false);
+
+	WifiEvents.onStationDisconnect(connectFail);
+	WifiEvents.onStationGotIP(gotIP);
+
+	auto timer = new SimpleTimer;
```

Comment:
@mikee47 When you are ready with this PR make sure to remove testing/debug code. Otherwise new developers will start copying it all over the place and think that this is an important part of UPnP, SSDP, etc :)